File: Template/TemplateFiles/java/withoutJNI/ReversibleDigitalInput.java
package $javapackage$;

import edu.wpi.first.wpilibj.DigitalInput;
import edu.wpi.first.wpilibj.SensorBase;

public class ReversibleDigitalInput extends SensorBase {
  private DigitalInput m_digitalInput;
  private boolean m_reverse = true;
  
  public ReversibleDigitalInput(int channel) {
    m_digitalInput = new DigitalInput(channel);
  }
  
  public void free() {
    m_digitalInput.free();
  }
  
  public void setReverseInput(boolean reverse) {
    m_reverse = reverse;
  }
  
  public boolean get() {
    boolean val = m_digitalInput.get();
    return m_reverse ? !val : val;
  }
}

File: Template/TemplateFiles/withoutDriver/ReversibleDigitalInput.cpp
#include "ReversibleDigitalInput.h"

namespace $vendorreplace$ {
  
using namespace frc;

ReversibleDigitalInput::ReversibleDigitalInput(int channel) : m_reverse(true) {
  m_digitalInput = std::make_unique<DigitalInput>(channel);
}

void ReversibleDigitalInput::SetReverseInput(bool reverse) {
  m_reverse = reverse;
}

bool ReversibleDigitalInput::Get() {
  bool val = m_digitalInput->Get();
  return m_reverse ? !val : val;
}

}
